lib/rubocop/cop/sorbet/t_enum/forbid_comparable_t_enum.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Sorbet
+      # Disallow including the `Comparable` module in `T::Enum`.
+      #
+      # @example
+      #
+      #  # bad
+      #  class Priority < T::Enum
+      #    include Comparable
+      #
+      #    enums do
+      #      High = new(3)
+      #      Medium = new(2)
+      #      Low = new(1)
+      #    end
+      #
+      #    def <=>(other)
+      #      serialize <=> other.serialize
+      #    end
+      #  end
+      class ForbidComparableTEnum < Base
+        include TEnum
+
+        MSG = "Do not use `T::Enum` as a comparable object because of significant performance overhead."
+
+        RESTRICT_ON_SEND = [:include, :prepend].freeze
+
+        # @!method include_comparable?(node)
+        def_node_matcher :include_comparable?, <<~PATTERN
+          (send nil? :include (const nil? :Comparable))
+        PATTERN
+
+        # @!method prepend_comparable?(node)
+        def_node_matcher :prepend_comparable?, <<~PATTERN
+          (send nil? :prepend (const nil? :Comparable))
+        PATTERN
+
+        def on_send(node)
+          return unless in_t_enum_class?
+          return unless include_comparable?(node) || prepend_comparable?(node)
```

Comment:
I think you can simplify the node matcher, see https://docs.rubocop.org/rubocop-ast/node_pattern.html#for-or-union.

Something like this:

```rb
        def_node_matcher :include_comparable?, <<~PATTERN
          (send nil? {:include | :prepend} (const nil? :Comparable))
        PATTERN
```

Reply:
Oh cool, thank you!

Reply:
Done!